packages/backend/src/api/controllers/tvl/DetailedTvlController.ts

```diff
@@ -0,0 +1,94 @@
+import { Logger } from '@l2beat/shared'
+import { DetailedTvlApiResponse, UnixTime } from '@l2beat/shared-pure'
+
+import { ReportProject } from '../../../core/reports/ReportProject'
+import { AggregatedReportRepository } from '../../../peripherals/database/AggregatedReportRepository'
+import { AggregatedReportStatusRepository } from '../../../peripherals/database/AggregatedReportStatusRepository'
+import { ReportRepository } from '../../../peripherals/database/ReportRepository'
+import { ReportStatusRepository } from '../../../peripherals/database/ReportStatusRepository'
+import { getHourlyMinTimestamp } from '../utils/getHourlyMinTimestamp'
+import { getSixHourlyMinTimestamp } from '../utils/getSixHourlyMinTimestamp'
+import {
+  groupByProjectIdAndAsset,
+  groupByProjectIdAndTimestamp,
+} from './detailedTvl'
+import { generateDetailedTvlApiResponse } from './generateDetailedTvlApiResponse'
+
+export class DetailedTvlController {
+  constructor(
+    private readonly reportStatusRepository: ReportStatusRepository,
+    private readonly aggregatedReportRepository: AggregatedReportRepository,
+    private readonly aggregatedReportStatusRepository: AggregatedReportStatusRepository,
+    private readonly reportRepository: ReportRepository,
+    private readonly projects: ReportProject[],
+    private readonly logger: Logger,
+  ) {
+    this.logger = this.logger.for(this)
+  }
+
+  async getDetailedTvlApiResponse(): Promise<
+    DetailedTvlApiResponse | undefined
+  > {
+    const [latestReportUpdateTime, latestAggregateReportUpdateTime] =
+      await Promise.all([
+        this.reportStatusRepository.findAnyLatestTimestamp(),
```

Comment:
shouldn't we find the minimum here? so we get latest timestamp on which all types are synced?

Reply:
There are different updaters for EBV, NMV and CBV

Reply:
They run in parallel

Reply:
We have to find those in the optimistic way anyway. Also, it would be great to build something to glue together fetching and setting safe sync heights, so we would prevent huge combined traversal query like `amount of chains * amount of supported assets' types on chains`